carbonate/sync.py

```diff
@@ -116,42 +123,61 @@ def heal_metric(source, dest, start_time=0, end_time=None, overwrite=False,
 
 
 def run_batch(metrics_to_sync, remote, local_storage, rsync_options,
-              remote_ip, dirty, lock_writes=False, overwrite=False):
-    staging_dir = mkdtemp(prefix=remote_ip)
-    sync_file = NamedTemporaryFile(delete=False)
-
-    metrics_to_heal = []
-
-    staging = "%s/" % (staging_dir)
-
-    for metric in metrics_to_sync:
-        staging_file = "%s/%s" % (staging_dir, metric)
-        local_file = "%s/%s" % (local_storage, metric)
-        metrics_to_heal.append((staging_file, local_file))
-
-    sync_file.write(("\n".join(metrics_to_sync)).encode())
-    sync_file.flush()
-
-    rsync_start = time()
-
-    sync_from_remote(sync_file, remote, staging, rsync_options)
-
-    rsync_elapsed = (time() - rsync_start)
-
-    merge_elapsed = sync_batch(metrics_to_heal, lock_writes=lock_writes,
-                               overwrite=overwrite)
-
-    total_time = rsync_elapsed + merge_elapsed
-
-    print("    --------------------------------------")
-    print("    Rsync time: %ss" % rsync_elapsed)
-    print("    Merge time: %ss" % merge_elapsed)
-    print("    Total time: %ss" % total_time)
-
-    # Cleanup
-    if dirty:
-        print("    dirty mode: left temporary directory %s" % staging_dir)
+              remote_ip, dirty, lock_writes=False, overwrite=False,
+              tmpdir=None, rsync_max_retries=3, rsync_retries_interval=5):
+    try:
+        staging_dir = mkdtemp(prefix=remote_ip, dir=tmpdir)
+    except OSError as e:
+        logging.error('Failed to create rsync staging dir %s: %s' %
+            (e.filename, e.strerror))
```

Comment:
Almost. Lint is not happy - 
carbonate/sync.py:132:13: E128 continuation line under-indented for visual indent